http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -88,8 +88,8 @@ final case class ClusterRequest(labels: LabelMap = Map.empty,
                                 autopauseThreshold: Option[Int] = None,
                                 defaultClientId: Option[String] = None,
                                 jupyterDockerImage: Option[ContainerImage] = None,
-                                rstudioDockerImage: Option[String] = None,
-                                welderDockerImage: Option[String] = None,
+                                rstudioDockerImage: Option[ContainerImage] = None,
```

Comment:
I'd prefer to make `jupyterDockerImage` and `rstudioDockerImage` into a sum type. Just so it's obvious that it's either or

Reply:
From a Leo perspective you can actually have both running at the same time (they will run in different containers and listen at different proxy URLs). Terra UI may enforce one or the other though. We need to think about the exact Terra UI behavior.

Reply:
gotcha. And we're not running welder when jupyter is not up right?

Reply:
Well right now it always launches Jupyter. If you don't specify a `jupyterDockerImage` in the request it uses the one in config. We probably want to retain that behavior for backwards compatibility. So the welder logic is unchanged as well.

`rstudioDockerImage` is truly optional -- there is no default and it doesn't influence Jupyter/welder.

Reply:
Actually I might change this behavior.. need to discuss with Adrian.